### ISSUE TITLE
[Features request] Add comments to Anki cards

### ISSUE THREAD

**Issue by kevinpolisano** [TowelSniffer/Anki-Chess-2.0#28]
Hello,

Thank you for this amazing plugin!

To complete my question on YT, could you add an option to display {comments} which are in the PGN file?

I would like to adapt [this method](https://nickguyrees.medium.com/what-does-it-take-to-get-good-at-chess-my-workflow-to-break-through-to-2100-elo-3f0be1f1bdc) in a more dynamic way thanks to your plugin. It would be great if I could mention somewhere my mistakes/blunders (in Front or Back of the card) and for instance mention the URI link to the Obsidian note of the whole game.

I tried to change the css with 80vh but then I cannot see the board entirely.

**Comment by debatepro** [comment]
If you select the Show Answer button at the bottom of the card you are sent to the version with the comments. Although, is difficult to view the comments if they span the visible area of the text box because there is no scroll bar. A scroll bar would be sufficient for me, given the comments from the PNG are on the review page. 
![ShowAnswer](https://github.com/TowelSniffer/Anki-Chess-2.0/assets/3801187/20ee783c-dfe4-40ee-830a-acbb3e42fc36)

**Comment by kevinpolisano** [comment]
Yes but I would like to be able to ask questions on the front, to orientate thoughts about the position.

**Comment by alexxthiery** [comment]
Thanks for the great pluggin!

I also agree with @kevinpolisano -- this would definitely be useful to have the option to ask questions on the front.

**Comment by TowelSniffer** [comment]
Hi, sorry for the late response here. So having an extra field for text certainly isn't impossible, but I ended up giving up on it due to it messing with how the cards rendered on ankidroid. It's just really annoying how inconsistent ankidroid, anki desktop and anki mobile are with rendering HTML. I might try again though as a few people have asked for the ability to add text.

**Comment by Amar0ks** [comment]
Any news about that =)?

**Comment by TowelSniffer** [comment]
So i just pushed a release there that should make the boardMoves box auto adjust in height to fit the PGN. I wish to avoid having comments on the frontside as I want the design to remain simple, however this should make the backside more usable, especially for longer PGNs/Comments. Let me know your thought.

**Comment by TowelSniffer** [comment]
last release includes a text field. It currently doesn't support formatting so would be limited to a line of text. I feel this is fine, but let me know if anyone has thoughts/suggestions regarding it. Text Isn't something I use with my cards.

As for the initial comment regarding displaying comments, this  feels quite doable now. The Idea I am thinking is having a box underneath that contains the comment for the last two moves played if they exists. So for black and whites move.  I feel like the initial request is satisfied by having a text Field though, so unless this is something that would actually be used id rather work on something else.

**Comment by kevinpolisano** [comment]
Hi !

Can you explain how to use the text field please?

Thanks!

**Comment by TowelSniffer** [comment]
If you're updated it should be as simple as adding text to the textField

**Comment by TowelSniffer** [comment]
When you import the latest template, it will likely create a second anki chess template as it won't be compatible with the old one

**Comment by kevinpolisano** [comment]
Ah OK it is "Chess 2.0+" 👍 

> The Idea I am thinking is having a box underneath that contains the comment for the last two moves played if they exists. So for black and whites move

Yes it would be perfect! Thanks in advance!

**Comment by TowelSniffer** [comment]
Ok last release is a fairly big redesign for comments. I think it looks pretty good. I added a fontSize option for a bit of customizability and it now renders HTML. It also auto displays any PGN comments as they appear. 

this is how it looks in the viewer

https://github.com/user-attachments/assets/93ad8e06-d426-4c61-b1a7-f33b3f4ba161


I have given up on mlieberts viewer and gone with my own design. It's a lot easier to work with and allowed me to add a lot of things I have wanted to add for a while, however some features are missing. The big one being you can no longer see the entire PGN. Instead moves are visually indicated with arrows. I personally prefer this but for those who really want the full PGN displayed you will either have to stick with older releases or wait until I add the feature in the future.

**Comment by TowelSniffer** [comment]
@kevinpolisano 

- So the arrows will only show alternate correct moves on the front side after the move was played. So not the current correct move. there is `var disableArrows = false` at the top of the template if you want to disable this. On the backside I haven't included this option as it's currently the only way to explore the PGN. 
- the template will import a note with the PGN (it now imports two note types for regular and flipped cards). You do need to have something in the textField field comments to display. 
- Are you saying the board doesn't flash green on completion? If so it sounds like rather than updating you existing note type, it has instead imported the new one as a duplicate. Check you note templates. anki should summarize imports before they occur so you can review them when importing an apkg file. you can replace the fron at back sections manually with the code found [here](https://github.com/TowelSniffer/Anki-Chess-2.0/tree/main/Note%20template) to fix it

**Comment by kevinpolisano** [comment]
Ok for the board flashing in green it works now. But after that on the back side after the moves was played, the arrows appear this is OK, however the board is flipped.

For instance with this study below I do the moves with the blacks, but afterward the correct moves with the green arrows are shown as if I play with the whites.

`1. d4 f5 {test} 2. Bf4 Nf6 3. e3 g6 4. Nf3 Bg7 5. Bd3 O-O 6. c3 d6 7. Nbd2 Nc6 8. O-O (8. Qc2 Qe8 9. O-O-O e5 10. dxe5 dxe5 11. Bg5 e4 12. Bc4+ Kh8 13. Nd4 Ne5) 8... Qe8 9. Re1 e5 10. dxe5 dxe5 11. Bg5 e4 12. Bc4+ Kh8 13. Nd4 Ne5 *`

Also the comment `{test}` does not appear, and the `textField` is filled.

**Comment by TowelSniffer** [comment]
@kevinpolisano do you have [discord?](https://discord.gg/mcNx5YSN)

**Comment by kevinpolisano** [comment]
OK let's continue on Discord!
